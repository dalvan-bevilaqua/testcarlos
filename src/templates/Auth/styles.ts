import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.primary.darker};
    height: 100%;
    width: 100%;
    display: grid;
    place-items: center;
    place-content: center;
    position: relative;
    min-height: 50rem;
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    z-index: ${theme.layers.base};
  `}
`

export const Form = styled.form`
  ${({ theme }) => css`
    background: ${theme.colors.gray.lightest};
    padding: ${theme.spacings.xlarge} ${theme.spacings.xsmall}
      ${theme.spacings.xmedium} ${theme.spacings.xsmall};
    border-radius: ${theme.border.rounded};
    text-align: center;
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    ${media.greaterThan('small')`
      width: 32rem;
    `}
  `}
`

export const InputGroup = styled.div`
  ${({ theme }) => css`
    text-align: start;
    width: 100%;
    ${media.greaterThan('small')`
      width: 28rem;
    `}
    margin: 0;
    margin-top: ${theme.spacings.small};
    display: grid;
    grid-gap: ${theme.spacings.xxsmall};
  `}
`

type ErrorMessageProps = {
  error?: boolean
}

export const ErrorMessage = styled.span<ErrorMessageProps>`
  ${({ theme, error }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 0.4rem;
    width: 100%;
    margin-top: ${theme.spacings.xxsmall};
    font-size: ${theme.font.sizes.xxsmall};
    color: ${theme.colors.tertiary.lighter};
    opacity: ${error ? 1 : 0};
    transition: all ${theme.transition.default};
  `}
`

export const ActionGroup = styled.div`
  ${({ theme }) => css`
    display: flex;
    margin-top: ${theme.spacings.xsmall};
    width: 28rem;
    justify-content: space-between;
    align-items: center;
  `}
`

export const Copyright = styled.span`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.small};
    font-size: ${theme.font.sizes.xxsmall};
    color: ${theme.colors.primary.lightest};
    text-align: center;
  `}
`
