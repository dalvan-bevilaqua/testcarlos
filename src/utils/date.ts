const months = [
  {
    key: '01',
    short: 'Jan',
    long: 'Janeiro'
  },
  {
    key: '02',
    short: 'Fev',
    long: 'Fevereiro'
  },
  {
    key: '03',
    short: 'Mar',
    long: 'MarÃ§o'
  },
  {
    key: '04',
    short: 'Abr',
    long: 'Abril'
  },
  {
    key: '05',
    short: 'Mai',
    long: 'Maio'
  },
  {
    key: '06',
    short: 'Jun',
    long: 'Junho'
  },
  {
    key: '07',
    short: 'Jul',
    long: 'Julho'
  },
  {
    key: '08',
    short: 'Ago',
    long: 'Agosto'
  },
  {
    key: '09',
    short: 'Set',
    long: 'Setembro'
  },
  {
    key: '10',
    short: 'Out',
    long: 'Outubro'
  },
  {
    key: '11',
    short: 'Nov',
    long: 'Novembro'
  },
  {
    key: '12',
    short: 'Dez',
    long: 'Dezembro'
  }
]

export function humanizeDate(date: string) {
  if (date) {
    try {
      const split = date.split('/')
      const day = split[0]
      const month = split[1]
      const current = months.find((item) => item.key === month)
      if (Number(day) <= 31 && current?.short) {
        return `${day} de ${current?.short}`
      }
    } catch {
      return ''
    }
  }
  return ''
}
