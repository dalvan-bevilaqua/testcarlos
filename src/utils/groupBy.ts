/* eslint-disable @typescript-eslint/no-explicit-any */

export default function groupBy(array: any, property: string) {
  return array.reduce(function (acc: any, obj: any) {
    const key = obj[property]
    if (!acc[key]) {
      acc[key] = []
    }
    acc[key].push(obj)
    return acc
  }, {})
}
