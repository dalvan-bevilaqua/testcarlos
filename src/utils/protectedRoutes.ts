import { GetServerSidePropsContext } from 'next'
import { getSession } from 'next-auth/client'

async function protectedRoutes(context: GetServerSidePropsContext) {
  const session = await getSession(context)

  const expires = (session?.expires || 0) < Date.now()

  if (!session || expires) {
    context.res.setHeader('Location', `/login`)

    context.res.statusCode = 302
  }

  return session
}

export default protectedRoutes
