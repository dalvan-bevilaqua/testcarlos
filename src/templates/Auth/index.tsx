import { FormEvent, useState, useEffect } from 'react'

import Link from 'next/link'
import Button from 'components/Button'
import HoverLink from 'components/HoverLink'
import Logo from 'components/Logo'
import TextField from 'components/TextField'
import * as S from './styles'
import Loading from 'components/Loading'
import { signIn } from 'next-auth/client'
import { useRouter } from 'next/dist/client/router'
import Waves from 'components/Waves'

const Auth = () => {
  const [error, setError] = useState<string | undefined>()
  const [loading, setLoading] = useState<boolean | undefined>(true)
  const routes = useRouter()
  const { push } = routes

  const [values, setValues] = useState({ email: '', password: '' })

  const handleInput = (field: string, value: string) => {
    setValues((s) => ({ ...s, [field]: value }))
  }

  useEffect(() => {
    setLoading(false)
    setError('')
  }, [])

  const handleSubmit = async (event: FormEvent) => {
    event.preventDefault()

    setLoading(true)
    !!error && setError('')

    const result = await signIn('credentials', {
      email: values.email,
      password: values.password,
      redirect: false
    })

    setLoading(false)

    if (result?.error) {
      setError(result?.error)
      return
    }

    if (result?.ok) {
      push('/')
      return
    }
  }

  return (
    <S.Wrapper>
      <S.Content>
        <S.Form onSubmit={handleSubmit}>
          <Logo />
          <S.InputGroup>
            <TextField
              error={!!error}
              label="E-mail:"
              type="email"
              required
              aria-label="email"
              onInputChange={(v) => handleInput('email', v)}
            />
            <TextField
              error={!!error}
              label="Senha:"
              type="password"
              required
              aria-label="password"
              onInputChange={(v) => handleInput('password', v)}
            />
          </S.InputGroup>
          <S.ErrorMessage error={!!error}>
            <span>{error}</span>
          </S.ErrorMessage>
          <S.ActionGroup>
            <HoverLink>
              <Link href="/forgot-password" passHref>
                Esqueci a Senha
              </Link>
            </HoverLink>
            <Button color="darker" size="medium" type="submit">
              <>{loading ? <Loading /> : <span>LOGIN</span>}</>
            </Button>
          </S.ActionGroup>
        </S.Form>
        <S.Copyright>
          examinaBem Â© 2022 Todos os direitos reservados.
        </S.Copyright>
      </S.Content>
      <Waves />
    </S.Wrapper>
  )
}

export default Auth
