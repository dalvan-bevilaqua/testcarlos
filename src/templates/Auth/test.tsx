import '@testing-library/jest-dom'
import { render } from 'utils/testUtils'
import Auth from '.'

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter')

const query = {}

useRouter.mockImplementation(() => ({
  query
}))

describe('<Auth />', () => {
  it('should render Auth with Logo', () => {
    const { getByText } = render(<Auth />)

    expect(getByText('examina')).toBeInTheDocument()
    expect(getByText('bem')).toBeInTheDocument()
  })
  it('should render Auth with inputs', () => {
    const { getByLabelText } = render(<Auth />)

    expect(getByLabelText('email')).toBeInTheDocument()
    expect(getByLabelText('password')).toBeInTheDocument()
  })

  it('should render Auth with forgot password link', () => {
    const { getByRole } = render(<Auth />)

    expect(getByRole('link')).toHaveAttribute('href', '/forgot-password')
  })

  it('should render Auth with button', () => {
    const { getByRole } = render(<Auth />)
    expect(getByRole('button')).toBeInTheDocument()
  })

  it('should match with snapshot ', () => {
    const { container } = render(<Auth />)

    expect(container).toMatchSnapshot()
  })
})
