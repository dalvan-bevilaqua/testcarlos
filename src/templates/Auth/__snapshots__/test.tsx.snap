// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> should match with snapshot  1`] = `
<div>
  <div
    class="sc-iCoGMd becHcS"
  >
    <div
      class="sc-fujyAs bPPkGT"
    >
      <form
        class="sc-pNWdM gyaqlL"
      >
        <div
          class="sc-dlnjwi jEqKCX"
        >
          <span>
            examina
          </span>
          <span>
            bem
          </span>
        </div>
        <div
          class="sc-jrsJWt gYdAtG"
        >
          <div
            class="sc-hKFxyN kNAsuG"
          >
            <label
              class="sc-gKAaRy cnjCFo"
            >
              E-mail:
            </label>
            <div
              class="sc-eCApnc jFFUJS"
            >
              <input
                aria-label="email"
                class="sc-jSFjdj gWBhyj"
                required=""
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class="sc-hKFxyN kNAsuG"
          >
            <label
              class="sc-gKAaRy cnjCFo"
            >
              Senha:
            </label>
            <div
              class="sc-eCApnc jFFUJS"
            >
              <input
                aria-label="password"
                class="sc-jSFjdj gWBhyj"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <span
          class="sc-kEqXSa ktkCOb"
        >
          <span />
        </span>
        <div
          class="sc-iqAclL fYHCGr"
        >
          <div
            class="sc-gtsrHT hlpuIv"
            color="default"
          >
            <a
              href="/forgot-password"
            >
              Esqueci a Senha
            </a>
          </div>
          <button
            class="sc-bdnxRM fSWHnF"
            color="darker"
          >
            <span>
              LOGIN
            </span>
          </button>
        </div>
      </form>
      <span
        class="sc-crzoAE gnnPHJ"
      >
        examinaBem Â© 2022 Todos os direitos reservados.
      </span>
    </div>
    <div
      class="sc-bqGGPW gydpLF"
    >
      <svg
        fill="none"
        viewBox="0 0 508 212"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 .7c5.5 1.4 11 3 16.5 4.8 69.8 22.6 125.7 68.3 161.8 126 31.3 50 92.3 73.4 149 57 42.6-12.3 89.2-12.4 134.6 2.2 16.3 5.3 31.7 12.2 45.9 20.6H0V.7z"
          fill="url(#paint0_linear)"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint0_linear"
            x1="-313.427"
            x2="663.545"
            y1="21.111"
            y2="285.901"
          >
            <stop
              offset=".44"
              stop-color="#8F8AFF"
            />
            <stop
              offset=".993"
              stop-color="#413AD1"
            />
          </lineargradient>
        </defs>
      </svg>
    </div>
    <div
      class="sc-ksluID eRQScO"
    >
      <svg
        fill="none"
        viewBox="0 0 509 212"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M508.1.7c-5.5 1.4-11 3-16.5 4.8-69.8 22.6-125.7 68.3-161.8 126-31.3 50-92.3 73.4-149 57-42.6-12.3-89.2-12.4-134.6 2.2C29.9 196 14.5 202.9.3 211.3h507.8V.7z"
          fill="url(#paint0_linear)"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint0_linear"
            x1="821.511"
            x2="-155.461"
            y1="21.111"
            y2="285.901"
          >
            <stop
              offset=".44"
              stop-color="#8F8AFF"
            />
            <stop
              offset=".993"
              stop-color="#413AD1"
            />
          </lineargradient>
        </defs>
      </svg>
    </div>
  </div>
</div>
`;
